server/src/main/java/org/elasticsearch/action/bulk/BulkRequestModifier.java

```diff
@@ -101,22 +102,33 @@ BulkRequest getBulkRequest() {
         }
     }
 
+    ActionListener<BulkResponse> wrapActionListenerIfNeeded(ActionListener<BulkResponse> actionListener) {
```

Comment:
It took me a little while to figure out what was going on here. It would probably be good to have a comment explaining why these two methods differ and why they are used. It might also be good to have a unit test that covers this version.

Reply:
I added documentation to both methods explaining the difference